test/e2e/catalog_e2e_test.go

```diff
@@ -916,11 +932,21 @@ var _ = Describe("Starting CatalogSource e2e tests", func() {
 				}
 			}
 			// update catalog source with annotation (to kick resync)
-			source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Get(context.Background(), source.GetName(), metav1.GetOptions{})
-			Expect(err).ShouldNot(HaveOccurred(), "error getting catalog source pod")
-			source.Annotations["testKey"] = genName("newValue")
-			_, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Update(context.Background(), source, metav1.UpdateOptions{})
-			Expect(err).ShouldNot(HaveOccurred(), "error updating catalog source pod with test annotation")
+			Eventually(func() error {
+				source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Get(context.Background(), source.GetName(), metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+
+				if source == nil {
+					return errors.New("could not get catalog source")
+				}
```

Comment:
What I had in mind was `source.Annotations = make(map[string]string)` but I won't hold up this PR as it fixes a real deflating style of e2e flake.